lib/src/firebase_storage_mocks_base.dart

```diff
@@ -9,26 +9,26 @@ class MockFirebaseStorage extends Mock implements FirebaseStorage {
   final Map<String, Uint8List> storedDataMap = {};
 
   @override
-  StorageReference ref() {
+  Reference ref([String path]) {
     return MockStorageReference(this);
```

Comment:
This code will return a reference to the root, even if someone calls `ref('/some/path')`. This is unexpected and it will confuse users of firebase_storage_mocks. Instead, can you throw an exception if someone passes a path, something like "paths are not supported yet"? Since the implementation would be non-trivial (it should support nested paths) and require more unit tests, throwing an exception is sufficient for now. We can implement it in a separate PR.

> NEW: To match the Web SDK, calling ref() creates a new Reference at the bucket root, whereas an optional path (ref('/foo/bar.png')) can be used to create a Reference pointing at a specific location.

https://pub.dev/packages/firebase_storage/changelog#500-dev1